### PR TITLE
Error handling docs

### PR DESCRIPTION
**Pull request by Rich-Harris**
Extends #7566

### CODE REVIEW

File: documentation/docs/30-advanced/25-errors.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
+---
+title: Errors
+---
+
+Errors are an inevitable fact of software development. SvelteKit handles errors differently depending on where they occur, what kind of errors they are, and the nature of the incoming request.
+
+### Error objects
+
+SvelteKit doesn't expose errors to users directly, Instead, errors are represented as simple `{ message: string }` objects by default.
```
Comment (benmccann) [inline documentation/docs/30-advanced/25-errors.md:9]: I'm guessing the errors talked about here are from `throw new Error` statements, so maybe add "thrown" to clarify (vs let's say a 500 which is an error but not `Error`). Also, there's a comma here that I think should be a period
```suggestion
SvelteKit doesn't expose thrown errors to users directly. Instead, errors are represented as simple `{ message: string }` objects by default.
```

**Review by benmccann — COMMENTED** [review]
I think this is a nice addition!

File: documentation/docs/30-advanced/25-errors.md (hunk 1 above)
Comment (dummdidumm) [reply documentation/docs/30-advanced/25-errors.md:9]: Trying to square the circle:

```suggestion
SvelteKit distinguishes between two kinds of errors - expected and unexpected ones. They are explained in more detail in the following sections. Both are represented as simple `{ message: string }` objects by default.
```

File: documentation/docs/30-advanced/25-errors.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+An _unexpected_ error is any other exception that occurs while handling a request. Since these can contain sensitive information, unexpected error messages and stack traces are not exposed to users.
+
+By default, unexpected errors are printed to the console (or, in production, your server logs), while the error that is exposed to the user has a generic shape:
+
+```json
+{ "message": "Internal Error" }
+```
+
+You can add your own error handling — for example, sending errors to a reporting service, or returning a custom error object — using the [`handleError`](/docs/hooks#shared-hooks-handleerror) hook.
```
Comment (dummdidumm) [inline documentation/docs/30-advanced/25-errors.md:80]:
```suggestion
Unexpected errors will go through the [`handleError`](/docs/hooks#shared-hooks-handleerror) hook, where you can add your own error handling — for example, sending errors to a reporting service, or returning a custom error object. 
```

File: documentation/docs/30-advanced/25-errors.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+		<title>%sveltekit.error.message%</title>
+	</head>
+	<body>
+		<h1>My custom error page</h1>
+		<p>Status: %sveltekit.status%</p>
+		<p>Message: %sveltekit.error.message%</p>
+	</body>
+</html>
+```
+
```
Comment (dummdidumm) [inline documentation/docs/30-advanced/25-errors.md:102]:
```suggestion

SvelteKit will replace `%sveltekit.status%` and `%sveltekit.error.message%` with their corresponding values.

```

File: documentation/docs/30-advanced/25-errors.md (hunk 1 above)
Comment (Rich-Harris) [reply documentation/docs/30-advanced/25-errors.md:9]: Yeah I like that. Slightly more compact version — I think the 'Expected errors' heading is close enough that we can omit the 'explained below' bit:

```suggestion
SvelteKit distinguishes between expected and unexpected errors, both of which are represented as simple `{ message: string }` objects by default.
```

File: documentation/docs/30-advanced/25-errors.md (hunk 2 above)
Comment (Rich-Harris) [reply documentation/docs/30-advanced/25-errors.md:80]: You can add code blocks inside suggestions, you just need to use four backticks on the outer fence (````` ````suggestion `````)

````suggestion
Unexpected errors will go through the [`handleError`](/docs/hooks#shared-hooks-handleerror) hook, where you can add your own error handling — for example, sending errors to a reporting service, or returning a custom error object. 

```js
/// file: src/hooks.server.js
// @errors: 2322 2571
// @filename: ambient.d.ts
const Sentry: any;

// @filename: index.js
// ---cut---
/** @type {import('@sveltejs/kit').HandleServerError} */
export function handleError({ error, event }) {
// example integration with https://sentry.io/
Sentry.captureException(error, { event });

return {
message: 'Whoops!',
code: error.code ?? 'UNKNOWN'
};
}
```
````

File: documentation/docs/30-advanced/25-errors.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+// ---cut---
+/** @type {import('@sveltejs/kit').HandleServerError} */
+export function handleError({ error, event }) {
+// example integration with https://sentry.io/
+Sentry.captureException(error, { event });
+
+return {
+message: 'Whoops!',
+code: error.code ?? 'UNKNOWN'
+};
```
Comment (Rich-Harris) [inline documentation/docs/30-advanced/25-errors.md:98]: whoops i messed up the indentation somehow

```suggestion
	// example integration with https://sentry.io/
	Sentry.captureException(error, { event });

	return {
		message: 'Whoops!',
		code: error.code ?? 'UNKNOWN'
	};
```

**Pull request merged**